holoviews/element/graphs.py

```diff
@@ -124,17 +127,37 @@ def __init__(self, data, **params):
                                             node_info.dimension_values(d),
                                             vdim=True)
             self._nodes = nodes
-        if self._edgepaths:
-            mismatch = []
-            for kd1, kd2 in zip(self.nodes.kdims, self.edgepaths.kdims):
-                if kd1 != kd2:
-                    mismatch.append('%s != %s' % (kd1, kd2))
-            if mismatch:
-                raise ValueError('Ensure that the first two key dimensions on '
-                                 'Nodes and EdgePaths match: %s' % ', '.join(mismatch))
+        self._validate()
         self.redim = redim_graph(self, mode='dataset')
 
 
+    def _validate(self):
+        if self._edgepaths is None:
+            return
+        mismatch = []
+        for kd1, kd2 in zip(self.nodes.kdims, self.edgepaths.kdims):
+            if kd1 != kd2:
+                mismatch.append('%s != %s' % (kd1, kd2))
+        if mismatch:
+            raise ValueError('Ensure that the first two key dimensions on '
+                             'Nodes and EdgePaths match: %s' % ', '.join(mismatch))
+        npaths = len(self._edgepaths.data)
+        nedges = len(self)
+        if nedges != npaths:
+            mismatch = True
+            if npaths == 1:
+                edges = self.edgepaths.split()[0]
+                vals = edges.dimension_values(0)
+                npaths = len(np.where(np.isnan(vals))[0])
+                if not np.isnan(vals[-1]):
+                    npaths += 1
+                mismatch = npaths != nedges
+            if mismatch:
+                raise ValueError('Ensure that the number of edges supplied to '
+                                 'the Graph matches the number of edgepaths: '
+                                 'expected %d, found %d' % (nedges, npaths))
```

Comment:
Probably clearer as ``Ensure that the number of edges supplied to the Graph (%d) matches the number of edgepaths (%d)' % (nedges, npaths)